NumLib/ODESolver/ConvergenceCriterionPerComponentResidual.cpp

```diff
@@ -35,6 +35,30 @@ ConvergenceCriterionPerComponentResidual::
         OGS_FATAL("The given tolerances vector is empty.");
 }
 
+
+void ConvergenceCriterionPerComponentResidual::checkDeltaX(
+    const GlobalVector& minus_delta_x, GlobalVector const& x)
+{
+    if ((!_dof_table) || (!_mesh))
+        OGS_FATAL("D.o.f. table or mesh have not been set.");
+
+    for (unsigned global_component = 0; global_component < _abstols.size();
+         ++global_component)
+    {
+        // TODO short cut if tol <= 0.0
+        auto error_dx = norm(minus_delta_x, global_component, _norm_type,
+                             *_dof_table, *_mesh);
+        auto norm_x =
+            norm(x, global_component, _norm_type, *_dof_table, *_mesh);
+
+        INFO(
+            "Convergence criterion, component %u: |dx|=%.4e, |x|=%.4e, "
```

Comment:
yes and it's same as other outputs in convergence criteria classes. `%g` is possible but the output format can change depending on the given value. `%f` should not be used, because it is fixed digits output and we can have various magnitudes of values, e.g. from 1e-18 to 1e+6